x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -352,51 +323,38 @@ public void prepareIndexIfNeededThenExecute(final Consumer<Exception> consumer,
                         + "Security features relying on the index will not be available until the upgrade API is run on the index");
             } else if (indexState.indexExists() == false) {
                 assert indexState.concreteIndexName != null;
-                logger.info("security index does not exist. Creating [{}] with alias [{}]", indexState.concreteIndexName, this.aliasName);
-                final byte[] mappingSource = mappingSourceSupplier.get();
-                final Tuple<String, Settings> mappingAndSettings = parseMappingAndSettingsFromTemplateBytes(mappingSource);
-                CreateIndexRequest request = new CreateIndexRequest(indexState.concreteIndexName)
-                        .alias(new Alias(this.aliasName))
-                        .mapping(mappingAndSettings.v1())
-                        .waitForActiveShards(ActiveShardCount.ALL)
-                        .settings(mappingAndSettings.v2());
+                logger.info("security index does not exist, creating [{}] with alias [{}]", indexState.concreteIndexName, this.aliasName);
+
+                // `TransportCreateIndexAction` will automatically apply the right mappings, settings and aliases, so none
+                // of that needs to be specified here.
+                CreateIndexRequest request = new CreateIndexRequest(indexState.concreteIndexName).waitForActiveShards(ActiveShardCount.ALL);
```

Comment:
Do we need to validate the mappings are good to go in cases where this node is not the master and there are mixed versions in the cluster?